src/jinja2_fragments/__init__.py

```diff
@@ -85,3 +85,57 @@ def render_block(
         return environment.concat(block_render_func(ctx))  # type: ignore
     except Exception:
         environment.handle_exception()
+
+
+def render_block_list(
+    environment: Environment,
+    template_name: str,
+    block_name_list: list[str],
+    *args: typing.Any,
+    **kwargs: typing.Any,
+) -> str:
+    """This returns one or more rendered template blocks as a string."""
+    if environment.is_async:
+        import asyncio
+
+        close = False
+
+        try:
+            loop = asyncio.get_running_loop()
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+            close = True
+
+        try:
+            raise NotImplementedError()
```

Comment:
Were you thinking on adding the async implementation later on?

Reply:
To be honest I didn't wanted to add an async implementation because I'm not familiar with `asyncio`, but since I'm going to make some changes I'll give it a shot